src/sentry/search/events/builder/metrics.py

```diff
@@ -166,17 +166,13 @@ def _get_on_demand_metric_spec(self, field: str) -> Optional[OnDemandMetricSpec]
                     "Must include on demand metrics type when querying on demand"
                 )
 
-            # Instead of calling OnDemandMetricSpec without a spec_version (defaulting to the least),
-            # we keep this logic here to make future spec versions easier to roll out since it will
-            # save us to look where to add this logic again
-            spec_version = OnDemandMetricSpecVersioning.get_query_spec_version(self.organization_id)
-            return OnDemandMetricSpec(
+            return fetch_on_demand_metric_spec(
```

Comment:
This change will make sure that the tests have the same function to get an `OnDemandMetricSpec` from a Relay metric spec.